include/gdsb/graph_io.h

```diff
@@ -68,13 +77,35 @@ constexpr bool full_graph = false;
 //! @param  subgraph        A subgraph to extract from the file. Use default or
 //!                         any other Subgraph object if not specified by
 //!                         ExtractSubgraph.
-template <typename Vertex, typename EmplaceF, bool IsDirected, bool IsWeighted, bool IsDynamic = false, typename Timestamp = uint64_t, bool ExtractSubgraph = false>
```

Comment:
a type safe graph description could go along

```
#include <iostream>

template <bool b> class Property{
public: static constexpr bool is(){return b;}
};

class Directed: private Property<true>{
public: static constexpr bool isDirected(){return Property::is();}  
};
class Undirected: private Property<false>{
public: static constexpr bool isDirected(){return Property::is();}  
};
class Weighted: private Property<true>{
public: static constexpr bool isWeighted(){return Property::is();}  
};
class Unweighted: private Property<false>{
public: static constexpr bool isWeighted(){return Property::is();}  
};
class Dynamic: private Property<true>{
public: static constexpr bool isDynamic(){return Property::is();}  
};
class Static: private Property<false>{
public: static constexpr bool isDynamic(){return Property::is();}  
};
template <typename Directed, 
          typename Weighted, 
          typename Dynamic> class InputGraph {
        public:
        static constexpr bool isDirected() {return Directed::isDirected();}
        static constexpr bool isWeighted() {return Weighted::isWeighted();}
        static constexpr bool isDynamic()  {return Dynamic::isDynamic();}
};

int main() {

    std::cout<< InputGraph<Directed, Weighted, Static>::isDirected()<<"\n";
    //std::cout<< InputGraph<Weighted, Directed, Static>::isDirected()<<"\n";   
}

```
or a condensed version of it, i know 'much ado about little' but not heavier in the end